autoblocks/_impl/testing/evaluators/has_all_substrings.py

```diff
@@ -9,25 +9,26 @@
 from autoblocks._impl.testing.models import Threshold
 
 
-class HasAllSubstrings(BaseTestEvaluator, Generic[TestCaseType, OutputType]):
+class HasAllSubstrings(BaseTestEvaluator, abc.ABC):
     """
     The HasAllSubstrings evaluator checks if the output contains all the expected substrings.
     Scores 1 if all substrings are present, 0 otherwise.
     The comparison is case-sensitive.
     """
 
-    id = "has-all-substrings"
+    @abc.abstractmethod
+    def test_case_mapper(self, test_case: Any) -> List[str]:
```

Comment:
i think we can keep the generics here as well

Reply:
I was getting a mypy error with the generics when I tried to add it in the test

error: Argument 1 of "output_mapper" is incompatible with supertype "ManualBattle"; supertype defines the argument type as "OutputType"  [override]
 note: This violates the Liskov substitution principle
note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

Reply:
hmm this isn't extending ManualBattle so maybe that error was coming from something else?

Reply:
It was coming from the test

```
    class Battle(ManualBattle):
        id = "battle"
        criteria = "Choose the best greeting."

        def output_mapper(self, output: str) -> str:
            return output

        def baseline_mapper(self, test_case: MyTestCase) -> str:
            return "goodbye world"
  ```

Reply:
yeah i am commenting on the HasAllSubstrings class here, but if this is only used in tests it's fine